### PR TITLE
Issue/42/setupextras: Closes #42

### LINKED ISSUE #42: Add code dependent install requirements to setup
the setup.py does not currently include dependencies for flexcode.  We should be able to set up code-specific dependencies in the install file, plus a "full" install that covers all dependencies.

### PR DESCRIPTION
**Pull request by sschmidt23**
This PR addresses issue #42 and adds extras_require options to setup.py. You can now pip install RAIL[Full] to get full dependence, or RAIL[flex] to install e.g. just the flexcode dependencies.  Future add on packages will have to add on to the extras_require as well as the `Full` to ensure that their dependencies install correctly.

PR runs ok on GitHub actions, so I think this is pretty straightforward.

### CODE REVIEW

**Comment by sschmidt23** [comment]
Should we maybe add a note to the README.md about install options?

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -26,9 +26,7 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install numpy scipy pyyaml h5py pandas tables pyarrow
-        pip install FlexCode[all]
-        pip install .
+        pip install .[Full]
         pip install flake8 pytest pytest-cov
         if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```
Comment (johannct) [inline .github/workflows/main.yml:31]: l.31 : do we want to keep the it ?

**Review by johannct — APPROVED** [review]
LGTM
We probably want to add already what is in the creation branch : 
corner
dill
flax==0.2.0
h5py
jax==0.1.75
jaxlib==0.1.52
numpy
pandas
scipy
scikit-learn
tensorflow==2.2.0
tensorflow-probability==0.10.1

`corner` is for a notebook though.....

**Comment by johannct** [comment]
> Should we maybe add a note to the README.md about install options?

Let's do that once we have a complete merge, and just before we are ready to tag RAIL

**Comment by sschmidt23** [comment]
I added the creation dependencies and created separate install options for creation and estimation.  Everything installs ok and all checks pass, so once I switch back the GitHub action trigger I will merge this.

**Pull request merged**